unit-tests/run-unit-tests.hs

```diff
@@ -318,7 +318,7 @@ test_package_db = IOTestCase "package_db" [dir] $ \wrapInterp -> do
               & last
                 -- "8.8.4"
         let pkgdb    = dir </> "dist-newstyle" </> "packagedb" </> ("ghc-" ++ ghcVersion)
-            ghc_args = ["-package-db=" ++ pkgdb]
+            ghc_args = ["-package-db=" ++ pkgdb,"-package-env","-"]
```

Comment:
I'm curious: did you enable `write-ghc-environment-variables` globally in your by any chance? It seems like this flag would only be necessary if you set `GHC_ENVIRONMENT` or have an environment file lying around.

Reply:
Ah, yeah, you're right. This change won't be needed.

Reply:
Thus reminds me that I really need to improve the documentation of how hint finds its package databases. The easy way to use hint is to build with `write-ghc-environment-files` enabled, this way hint can automatically find its package databases at runtime. The hard way is to figure out where cabal and stack hide their package databases, and to specify a path to them via these `-package-db` args.

Since this test is specifically about checking that hint can use this hard way, it makes sense to disable the environment file in this test using `-package-env -`. At some point it would be nice to also add a test for the easy way!

Reply:
documentation added in ff15999fee2241b41858907ae5645eb1e95dc176

test added in dad34b9685d14aea4335ba8eed63532eaaa4f384